src/coordinax/_src/d3/spheroidal.py

```diff
@@ -0,0 +1,135 @@
+"""Built-in vector classes."""
+
+__all__ = ["ProlateSpheroidalPos", "ProlateSpheroidalVel", "ProlateSpheroidalAcc"]
+
+from dataclasses import KW_ONLY
+from functools import partial
+from typing import final
+
+import equinox as eqx
+
+import quaxed.numpy as xp
+from dataclassish.converters import Unless
+from unxt import Quantity
+
+import coordinax._src.typing as ct
+from .base import AbstractAcc3D, AbstractPos3D, AbstractVel3D
+from coordinax._src.angle import Angle
+from coordinax._src.checks import (
+    check_greater_than_equal,
+    check_less_than_equal,
+    check_r_non_negative,
+)
+from coordinax._src.converters import converter_azimuth_to_range
+from coordinax._src.distance import AbstractDistance, Distance
+from coordinax._src.utils import classproperty
+
+
+@final
+class ProlateSpheroidalPos(AbstractPos3D):
+    """Prolate spheroidal coordinates as defined by Dejonghe & de Zeeuw 1988.
+
+    Parameters
+    ----------
+    mu : `coordinax.distance.Distance`
+        The spheroidal mu coordinate. This is called `lambda` by Dejonghe & de Zeeuw.
+    nu : `coordinax.distance.Distance`
+        The spheroidal nu coordinate.
+    phi : `coordinax.angle.Angle`
+        Azimuthal angle [0, 360) [deg] where 0 is the x-axis.
+    Delta : `coordinax.distance.Distance`
+        The focal length of the coordinate system.
+
+    """
+
+    mu: ct.BatchableArea = eqx.field(
+        converter=partial(Quantity["area"].from_, dtype=float)
+    )
+    r"""Spheroidal mu coordinate :math:`\mu \in [0,+\infty)` (called :math:`\lambda` in
+     some Galactic dynamics contexts)."""
+
+    nu: ct.BatchableArea = eqx.field(
+        converter=partial(Quantity["area"].from_, dtype=float)
+    )
+    r"""Spheroidal nu coordinate :math:`\lambda \in [-\infty,+\infty)`."""
+
+    phi: ct.BatchableAngle = eqx.field(
+        converter=Unless(
+            Angle, lambda x: converter_azimuth_to_range(Angle.from_(x, dtype=float))
+        )
+    )
+    r"""Azimuthal angle, generally :math:`\phi \in [0,360)`."""
+
+    _: KW_ONLY
+    Delta: ct.BatchableDistance = eqx.field(
```

Comment:
This can now be a `VectorAttribute`

Reply:
Slightly confused about this -- should it look like:
```python
Delta: Shaped[Quantity["length"], ""] = eqx.field(
    default=VectorAttribute(), repr=False
)
```
? Do I need to detect a missing value and raise an error myself?

Reply:
What's going wrong, exactly? Is there something happening in ``__init__``?

Reply:
Nothing going wrong per se. But if I try to initialize without specifying a value for `Delta`, it fails on one of the value checks instead of saying something nice like "You must provide a value for the focal length Delta":

```python
cx.ProlateSpheroidalPos(
    mu=Quantity([1, 2, 3, 4], "kpc2"),
    nu=Quantity([0.1, 0.2, 0.3, 0.4], "kpc2"),
    phi=Quantity([0, 1, 2, 3], "rad")
)
```
...
```python-traceback
TypeError: '<=' not supported between instances of 'Sentinel' and 'int'

```